whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java

```diff
@@ -0,0 +1,197 @@
+package net.ripe.db.whois.common.sso;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.jaxrs.json.JacksonJaxbJsonProvider;
+import com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider;
+import net.ripe.db.whois.common.sso.domain.MemberContactsResponse;
+import net.ripe.db.whois.common.sso.domain.ValidateTokenResponse;
+import org.apache.commons.lang.StringUtils;
+import org.glassfish.jersey.client.ClientProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.UNAUTHORIZED;
+
+@Component
+public class AuthServiceClient {
+
+    public static final String TOKEN_KEY = "crowd.token_key";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthServiceClient.class);
+
+    private static final String VALIDATE_PATH = "/authorisation-service/v2/authresource/validate";
```

Comment:
Should the base URL have `/authorisation-service/` in it? Then these relative paths are within the authorisation service.

Reply:
Thanks, it also makes the code more readable, i.e. we only need relative paths then within the API.